test/types/range/ferguson/range-intents.chpl

```diff
@@ -0,0 +1,184 @@
+// This test runs a series of test methods to verify
+// that range behaves similarly to int as far as
+// argument intents are concerned.
+
+config param testrange = true;
+
+proc make(x:int) {
+  if testrange then
+    return 1..x;
+  else
+    return x;
+}
+proc upper(x) { return x.high; }
+proc upper(x:int) { return x; }
+
+var start = make(10);
+var change = make(20);
+
+writeln(upper(start), " means a copy was made; ", upper(change), " means some sort of ref");
+
+var global = start;
+
+proc test_blank( arg ) {
+  global = change;
+  writeln("default ", upper(arg));
+}
+global = start;
+test_blank(global);
+
+proc test_const( const arg ) {
+  global = change;
+  writeln("const ", upper(arg));
+}
+global = start;
+test_const(global);
+
+proc test_in( in arg ) {
+  global = change;
+  writeln("in ", upper(arg));
+}
+global = start;
+test_in(global);
+
+proc test_const_in( const in arg ) {
+  global = change;
+  writeln("const in ", upper(arg));
+}
+global = start;
+test_const_in(global);
+
+proc test_const_ref( const ref arg ) {
+  global = change;
+  writeln("const ref ", upper(arg));
+}
+global = start;
+test_const_ref(global);
+
+proc test_ref( ref arg ) {
+  global = change;
+  writeln("ref ", upper(arg));
+}
+global = start;
+test_ref(global);
+
+proc range.test_this() {
+  global = change;
+  writeln("this ", upper(this));
+}
+proc int.test_this() {
+  global = change;
+  writeln("this ", upper(this));
+}
+global = start;
+global.test_this();
+
+proc const range.test_const_this() {
+  global = change;
+  writeln("const this ", upper(this));
+}
+proc const int.test_const_this() {
+  global = change;
+  writeln("const this ", upper(this));
+}
+global = start;
+global.test_const_this();
+
+/*proc in range.test_in_this() {
```

Comment:
Should this commented-out block still be here? If so, could you add a comment explaining why?

Reply:
We don't support 'in' or 'const in' as 'this' intents. (Why not? I dunno.) If we did, I'd want these functions uncommented to test it.